### PR TITLE
Fix asar loading which was broken due to node bootstrap refactoring

### PR DESCRIPTION
**Pull request by nitsakh**
`lib/internal/bootstrap_node.js` was broken down and refactored into different files.
This change addresses that and restores asar support for electron.

### CODE REVIEW

File: lib/internal/bootstrap/loaders.js [hunk 1]
```diff
@@ -99,7 +101,7 @@
     this.loading = false;
   }
 
-  NativeModule._source = getBinding('natives');
+  NativeModule._source = process.binding('natives');
```
Comment (alexeykuzmin) [inline lib/internal/bootstrap/loaders.js]: What't the difference between `getBinding` and `process.binding`?

File: lib/internal/bootstrap/loaders.js [hunk 2]
```diff
@@ -50,7 +50,9 @@
     enumerable: true,
     writable: false
   });
+  process.NativeModule = NativeModule;
 
+  var localGlobal = global
```
Comment (alexeykuzmin) [inline lib/internal/bootstrap/loaders.js]: `const` and a semicolon please =)

File: lib/internal/bootstrap/loaders.js (hunk 1 above)
Comment (nitsakh) [reply lib/internal/bootstrap/loaders.js]: `process.binding` is legacy, and as mentioned in the comment at the beginning of `loaders.js`, they have made changes for embedders.
I'm not exactly sure how we can use linkedbinding, but as little as I know, `process.binding` and `getBinding` should work the same way.

Comment (alexeykuzmin) [reply lib/internal/bootstrap/loaders.js]: Honestly, I don't any reasons to make this change. Let's remove it )

File: lib/internal/bootstrap/node.js [hunk 3]
```diff
@@ -24,6 +24,8 @@
 
     EventEmitter.call(process);
 
+    process.NativeModule = NativeModule;
```
Comment (alexeykuzmin) [inline lib/internal/bootstrap/node.js]: It looks like you're doing it twice )

Comment (alexeykuzmin) [reply lib/internal/bootstrap/node.js]: Let's leave it here, and remove from the "lib/internal/bootstrap/loaders.js".

**Comment by MylesBorins** [comment]
Is there anything going on here that we should look at upstream?

**Comment by alexeykuzmin** [comment]
@MylesBorins 
> Is there anything going on here that we should look at upstream?

I guess no, we just had to move some pieces to the new files.
But we will probably try to reduce the amount of changes needed to make our stuff work in the near future. Thank you!

**Pull request merged**